php/libraries/User.class.inc

```diff
@@ -66,26 +66,41 @@ class User extends UserPermissions
         $row = $DB->pselectRow($query, array('UID' => $username));
```

Comment:
sure, but i don't get any warning or error for if User::factory() is called without an argument. is that normal?